### PR TITLE
LizardFS: 3.11.3 -> 3.12.0, add Berkeley DB support

### PR DESCRIPTION
**Pull request by kliu128**
###### Motivation for this change

LizardFS 3.12.0 is out. Also, Berkeley DB is a nice name caching system that reduces the memory usage of the master server a little.

I have been using these commits for a while in a private fork of Nixpkgs.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by markuskowa — CHANGES_REQUESTED** [review]

File: pkgs/tools/filesystems/lizardfs/default.nix [hunk 1]
```diff
@@ -16,28 +18,37 @@
...
+    [ cmake db fuse asciidoc libxml2 libxslt docbook_xml_dtd_412 docbook_xsl
       zlib boost pkgconfig judy pam makeWrapper
     ];
 
+  patches = [
+    ./remove-download-external.patch
+  ];
+
+  postUnpack = ''
+    ls -lh
```
[the remark this answers is not in the document]
Comment (Pneumaticat) [reply pkgs/tools/filesystems/lizardfs/default.nix]: Yup!

File: pkgs/tools/filesystems/lizardfs/spdlog.nix [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{ fetchzip }:
```
[the remark this answers is not in the document]
Comment (Pneumaticat) [reply pkgs/tools/filesystems/lizardfs/spdlog.nix]: Agreed. I've updated the commit to do that.

File: pkgs/tools/filesystems/lizardfs/default.nix [hunk 3]
```diff
@@ -16,28 +18,44 @@
...
     rev = "v${version}";
-    sha256 = "1njgj242vgpdqb1di321jfqk4al5lk72x2iyp0nldy7h6r98l2ww";
+    sha256 = "0zk73wmx82ari3m2mv0zx04x1ggsdmwcwn7k6bkl5c0jnxffc4ax";
   };
 
-  buildInputs = 
-    [ cmake fuse asciidoc libxml2 libxslt docbook_xml_dtd_412 docbook_xsl
+  buildInputs =
+    [ cmake db fuse asciidoc libxml2 libxslt docbook_xml_dtd_412 docbook_xsl
       zlib boost pkgconfig judy pam makeWrapper
```
Comment (markuskowa) [inline pkgs/tools/filesystems/lizardfs/default.nix]: `cmake`, `pkgconfig`, and `makeWrapper` should go in `nativeBuildInputs`.

**Comment by markuskowa** [comment]
@GrahamcOfBorg build lizardfs

File: pkgs/tools/filesystems/lizardfs/default.nix (hunk 3 above)
Comment (Pneumaticat) [reply pkgs/tools/filesystems/lizardfs/default.nix]: Fixed!

**Comment by markuskowa** [comment]
@GrahamcOfBorg build lizardfs

**Review by markuskowa — APPROVED** [review]

**Pull request merged**